docs/tutorials/intro.md

```diff
@@ -223,12 +183,74 @@ fn main() {
 If all went well, running the app with `cargo run` should display:
 
 ```
-📅: 14 października 2020 13:21:00
+📅: 2020年10月14日 13:21:00
 ```
 
 Here's an internationalized date!
 
-*Notice:* Default `cargo run` builds and runs a `debug` mode of the binary. If you want to evaluate performance, memory or size of this example, use `cargo run --release`. Our example is also using `json` resource format. Generate the data in `bincode` for better performance.
+*Notice:* Default `cargo run` builds and runs a `debug` mode of the binary. If you want to evaluate performance, memory or size of this example, use `cargo run --release`. Our example is also using `json` resource format. Generate the data in `postcard` (and use `BlobDataProvider`) for better performance.
+
+## Using data from the filesystem
+
+If you have ICU4X data on the file system in a JSON format, it can be loaded via `FsDataProvider`:
+
+```toml
+[dependencies]
+icu = "0.5"
+icu_provider_fs = {version = "0.5" , features = ["deserialize_json"]}
+```
+
+```rs
+use icu_provider_fs::FsDataProvider;
+
+fn main() {
+    let _provider = FsDataProvider::try_new("/path/to/data")
+       .expect("Failed to initialize Data Provider.");
+}
+```
+
+The ICU4X repository has test data checked in tree in `provider/testdata/data/json`, however it is recommended one generate data on their own as described in the [next section](#generating data). Under the hood, `icu_testdata::get_provider()` is simply loading this data.
+
+## Generating data
+
+For production usage, it is better to generate your own data that is filtered to suit your needs.
+
+We're going to use [JSON CLDR](https://github.com/unicode-cldr/cldr-json) as our source data. JSON CLDR is an export of [CLDR data](http://cldr.unicode.org/index/downloads) into JSON maintained by Unicode.
+
+We are also going to use Unicode property data shipped as a zip file in the ICU4C release.
+
+The `datagen` component has a binary application which will fetch the CLDR data and generate ICU4X data out of it.
+
+```
+cd ~/projects/icu
+wget https://github.com/unicode-org/icu/releases/download/release-70-1/icuexportdata_uprops_full.zip
+unzip icuexportdata_uprops_full.zip
+git clone https://github.com/unicode-org/icu4x
+cd icu4x
+git checkout icu@0.5.0
+cargo run --bin icu4x-datagen -- \
+    --cldr-tag 40.0.0 \
+    --uprops-root ../icuexportdata_uprops_full/small \
+    --out ~/projects/icu/icu4x-data \
+    --all-keys --all-locales
```

Comment:
The uprops zip doesn't have to be manually downloaded anymore. 

Also, `--all-keys` doesn't work with ICU70 as the script extensions data is not included, so we should point this to either release-71-rc or release-71-1. 

```suggestion
git clone https://github.com/unicode-org/icu4x
cd icu4x
git checkout icu@0.5.0
cargo run --bin icu4x-datagen --features download -- \
    --cldr-tag 40.0.0 \
    --uprops-tag release-71-1 \
    --out ~/projects/icu/icu4x-data \
    --all-keys --all-locales
```

Reply:
Also fix up the explanation of the command below.

Reply:
And this also removes the `cd ~/projects/icu`, so don't tell the user to navigate there.

Reply:
There's still a cd up above